csrc/fusion_segmenter.cpp

```diff
@@ -1898,6 +1898,36 @@ void convertInputLogicalsToRoots(Fusion* fusion) {
       new_td = IrBuilder::create<TensorDomain>(
           new_logical_domain, tv->domain()->contiguity());
     }
+
+    // Remove reduction domains from new_td
+    std::vector<IterDomain*> no_red_alloc;
+    std::vector<std::optional<bool>> no_red_contiguity;
+    bool has_reduction = false;
+    for (size_t i : c10::irange(new_td->maybeAllocation().size())) {
```

Comment:
Good question. Just in case, we can check both `allocation()` and `logical()`.

Reply:
On the other hand, just checking `logical()` should be enough to determine if there is any reduction domain I guess, since we'd definitely not expect to find a reduction axis in the allocation domain if there are none in the logical domain.